### DISCUSSION TITLE
missing index.html for docs subfolder

### DISCUSSION THREAD

**Discussion by johnxba** [facebook/docusaurus#10536] [Q&A]
Just in case it helps someone: the first md file in a docs subfolder should have the same ```name```+```.md``` as the folder.

**Comment by Advait251206** [comment]
I don't think this is a general Docusaurus requirement.

Docusaurus does **not** require the first Markdown file inside a docs subfolder to have the same name as the folder in order to generate an `index.html`. Documentation pages are generated based on document IDs, routes, sidebar configuration, and front matter—not the filename matching the directory name.

If renaming the file fixed the issue in your case, it was likely due to your project's specific configuration (for example, autogenerated sidebars, `slug`, or other routing settings) rather than a built-in requirement.

For anyone running into a missing `index.html` for a docs subfolder, it's worth checking:

- Whether the folder contains at least one valid Markdown document.
- Whether the document is included in the generated sidebar (especially when using autogenerated sidebars).
- Whether `slug` or other front matter is changing the expected route.
- Whether a `_category_.json` or `_category_.yml` file is affecting the folder's generated route.
- Whether there are any build warnings or errors that indicate why the route wasn't generated.

So while your workaround may solve the issue for a particular setup, it shouldn't be considered a general rule for Docusaurus projects.

--

If this answer helped or pointed you in the right direction, I'd appreciate it if you could mark it as the accepted answer so it's easier for others with the same issue to find.

Also, if you found my contribution useful, I'd appreciate it if you could check out my GitHub profile, follow me, and star any repositories you find interesting.

GitHub: https://github.com/Advait251206
